mapeditor/mainwindow.cpp

```diff
@@ -412,9 +417,21 @@ bool MainWindow::openMap(const QString & filenameSelect)
 	
 	filename = filenameSelect;
 	initializeMap(controller.map()->version != EMapFormat::VCMI);
+
+	updateRecentMenu(filenameSelect);
+
 	return true;
 }
 
+void MainWindow::updateRecentMenu(const QString & filenameSelect) {
```

Comment:
Most Qt core classes implement COW, no need to pass them via const ref
```suggestion
void MainWindow::updateRecentMenu(QString filenameSelect)
{
```

Reply:
All functions in MainWindow.cpp pass QString by const reference. Might be out of scope of this PR to break consistency with the rest of the code. The intent is also to say that the QString is expected not to be modified, so it should be const.

Reply:
a good policy is to write new code properly, even if it "violates" style of the old code. Not insisting on changing it in this PR then, but let's prefer this approach in the future.

if you pass by value, original object isn't modified as well.

Reply:
I'll fix it for this PR